source/fuzz/shrinker.cpp

```diff
@@ -237,22 +242,30 @@ Shrinker::ShrinkerResultStatus Shrinker::Run(
     }
   }
 
-  // The output from the shrinker is the best binary we saw, and the
-  // transformations that led to it.
-  *binary_out = current_best_binary;
-  *transformation_sequence_out = current_best_transformations;
-
   // Indicate whether shrinking completed or was truncated due to reaching the
   // step limit.
+  //
+  // Either way, the output from the shrinker is the best binary we saw, and the
+  // transformations that led to it.
   assert(attempt <= step_limit_);
   if (attempt == step_limit_) {
     std::stringstream strstream;
     strstream << "Shrinking did not complete; step limit " << step_limit_
               << " was reached.";
     consumer_(SPV_MSG_WARNING, nullptr, {}, strstream.str().c_str());
```

Comment:
According to https://en.cppreference.com/w/cpp/io/basic_stringstream/str, `str()` returns a temporary object. Thus, `str().c_str()` results in a dangling pointer.

I've seen this bug somewhere else in the project as well. Maybe it would be better to make a separate PR for fixing it.

Reply:
I believe this is OK because the lifetime of a temporary object is dictated by the "full-expression" in which it is nested.

Here, the full-expression is:

```
consumer_(SPV_MSG_WARNING, nullptr, {}, strstream.str().c_str())
```

I believe there would only be UB if we did something like this instead:

```
auto temp = strstr.str().c_str();
consumer_(SPV_MSG_WARNING, nullptr, {}, temp)
```

Reply:
Yes, you are right! I didn't know that. In this case, I'm OK with the current approach.

Reply:
Thanks.